### PR TITLE
deprecate `--format` and `--list` in `into datetime`

### PR DESCRIPTION
**Pull request by amtoine**
related to
- https://discord.com/channels/601130461678272522/614593951969574961/1141009665266831470

# Description
this PR
- prints a colorful warning when a user uses either `--format` or `--list` on `into datetime`
- does NOT remove the features for now, i.e. the two options still work
- redirect to the `format date` command instead

i propose to
- land this now
- prepare a removal PR right after this
- land the removal PR in between 0.84 and 0.85

# User-Facing Changes
`into datetime --format` and `into datetime --list` will be deprecated in 0.85.

## how it looks
- `into datetime --list` in the REPL
```nushell
> into datetime --list | first
Error:   × Deprecated option
   ╭─[entry #1:1:1]
 1 │ into datetime --list | first
   · ──────┬──────
   ·       ╰── `into datetime --list` is deprecated and will be removed in 0.85
   ╰────
  help: see `format datetime --list` instead


╭───────────────┬────────────────────────────────────────────╮
│ Specification │ %Y                                         │
│ Example       │ 2023                                       │
│ Description   │ The full proleptic Gregorian year,         │
│               │ zero-padded to 4 digits.                   │
╰───────────────┴────────────────────────────────────────────╯
```

- `into datetime --list` in a script
```nushell
> nu /tmp/foo.nu
Error:   × Deprecated option
   ╭─[/tmp/foo.nu:4:1]
 4 │ #
 5 │ into datetime --list | first
   · ──────┬──────
   ·       ╰── `into datetime --list` is deprecated and will be removed in 0.85
   ╰────
  help: see `format datetime --list` instead


╭───────────────┬────────────────────────────────────────────╮
│ Specification │ %Y                                         │
│ Example       │ 2023                                       │
│ Description   │ The full proleptic Gregorian year,         │
│               │ zero-padded to 4 digits.                   │
╰───────────────┴────────────────────────────────────────────╯
```

- `help into datetime`

baz

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by amtoine** [comment]
> I like it but don't think we can move forward with it unless it respects `use_ansi_coloring`. If `use_ansi_coloring` is false, all colors need to be removed. Some people have terminals that don't support coloring, believe it or not.
> 
very good catch :clap: :clap: 
should be better with 848344c48 :+1:
 
> @sholderbach was also working on a `ShellWarning` system, although I'm not sure of it's status.
>
i'd be happy to use a better and more clever system :relieved: 
this is my quick and dirty proposal for now :wink:

**Comment by amtoine** [comment]
thanks to @kubouch expertise, i've switched to using `report_error` and a `GenericError` to have pretty errors with span information :yum:

**Review by fdncred — APPROVED** [review]
I think this is good for `--options` until we can get `ShellWarning` up and running, assuming we go that direction. I'm fine with landing this for now.

File: crates/nu-command/src/conversions/into/datetime.rs [hunk 1]
```diff
@@ -111,6 +113,17 @@ impl Command for SubCommand {
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
         if call.has_flag("list") {
+            report_error(
```
[the remark this answers is not in the document]
Comment (amtoine) [reply crates/nu-command/src/conversions/into/datetime.rs:116]: ooh i like that :smirk:

[the remark this answers is not in the document]
Comment (amtoine) [reply crates/nu-command/src/conversions/into/datetime.rs]: done in c9ffc5696 :muscle:

**Comment by amtoine** [comment]
( i force pushed immediately to amend https://github.com/nushell/nushell/pull/10017/commits/dc2f814a8ab01f0e473207376c78bda3aca9ec06 with a Clippy fix )

**Review by kubouch — APPROVED** [review]

**Comment by fdncred** [comment]
Let's move forward with this and see how it goes.

**Comment by jmoore34** [comment]
@amtoine My understanding is the --format flag of `into datetime` changes the input formatting, while `date format` only changes the output formatting. Feel free to correct me if I'm missing something.

https://github.com/amtoine/nushell/blob/c9ffc5696debd310c8aa4a81d85201fd5e7907ee/crates/nu-command/src/conversions/into/datetime.rs#L330C5-L334C71
```
// If input is not a timestamp, try parsing it as a string
    match input {
        Value::String { val, span } => {
            match dateformat {
                Some(dt) => match DateTime::parse_from_str(val, &dt.0)
```

From the documentation of [parse_from_str](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.parse_from_str):

> Parses a string from a user-specified format into a DateTime<FixedOffset> value.

**Comment by fdncred** [comment]
Oh, `input` format. That would make a lot of sense then, but the problem with no one knowing understanding what it did was a problem. LOL. Maybe we revert this and chose something other than `-f` and be more specific about it being and `input-format`.

**Comment by amtoine** [comment]
so do we revert this PR? before the release?

**Comment by fdncred** [comment]
Thanks for putting up a revert PR @amtoine. I'm still confused though because it seems that something isn't working right.
```nushell
'20220604' | into datetime -f '%Y%m%d'
```
@jmoore34 Shouldn't this work or am I still missing the point to `--format`?

**Comment by fdncred** [comment]
Maybe we need a better datetime parsing create. Seems silly that the first one would work but the last one wouldn't. I just took off the %z and +0000.
```nushell
❯ '20210227_135540+0000' | into datetime -f '%Y%m%d_%H%M%S%z'
Sat, 27 Feb 2021 13:55:40 +0000 (2 years ago)
❯ '20210227_135540' | into datetime -f '%Y%m%d_%H%M%S'
Error: nu::shell::cant_convert

  × Can't convert to could not parse as datetime using format '%Y%m%d_%H%M%S'.
   ╭─[entry #12:1:1]
 1 │ '20210227_135540' | into datetime -f '%Y%m%d_%H%M%S'
   ·                     ──────┬──────
   ·                           ╰── can't convert input is not enough for unique date and time to could not parse as datetime using format '%Y%m%d_%H%M%S'
   ╰────
  help: you can use `into datetime` without a format string to enable flexible parsing
```

**Comment by fdncred** [comment]
Seems like `dtparse` may be better? https://github.com/bspeice/dtparse

**Pull request merged**